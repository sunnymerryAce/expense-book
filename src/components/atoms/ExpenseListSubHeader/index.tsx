import React from "react";
import ListSubheader from "@material-ui/core/ListSubheader";
import useStyles from "./style";

interface ExpenseListSubHeaderProps {
  dateLabel: string;
}

const ExpenseListSubHeader: React.FC<ExpenseListSubHeaderProps> = (props) => {
  const classes = useStyles();

  return (
    <ListSubheader className={`ExpenseListSubHeader ${classes.root}`}>
      {props.dateLabel}
    </ListSubheader>
  );
};

export default ExpenseListSubHeader;
