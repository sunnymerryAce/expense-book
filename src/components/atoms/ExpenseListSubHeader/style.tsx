import { makeStyles, createStyles, Theme } from "@material-ui/core/styles";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      backgroundColor: theme.palette.primary.light,
      color: "#fff",
    },
  })
);

export default useStyles;
