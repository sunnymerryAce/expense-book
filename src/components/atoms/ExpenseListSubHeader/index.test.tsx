import React from "react";
import ExpenseListSubHeader from "./";
import { shallow } from "enzyme";

describe("<ExpenseListSubHeader />", () => {
  let container: any = null;
  const props = {
    dateLabel: "2020/05/01",
  };

  beforeEach(() => {
    container = shallow(<ExpenseListSubHeader {...props} />);
  });

  afterEach(() => {
    container.unmount();
    container = null;
  });

  test("should match the snapshot", () => {
    expect(container.html()).toMatchSnapshot();
  });

  test("should show proper text", () => {
    expect(container.find(".ExpenseListSubHeader").text()).toBe(
      props.dateLabel
    );
  });
});
