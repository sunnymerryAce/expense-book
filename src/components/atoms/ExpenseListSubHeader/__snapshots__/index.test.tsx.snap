// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExpenseListSubHeader /> should match the snapshot 1`] = `"<li class=\\"MuiListSubheader-root ExpenseListSubHeader makeStyles-root-1 MuiListSubheader-sticky MuiListSubheader-gutters\\">2020/05/01</li>"`;
